homeassistant/components/velbus/quality_scale.yaml

```diff
@@ -0,0 +1,70 @@
+rules:
+  # Bronze
+  action-setup: todo
+  appropriate-polling:
+    status: exempt
+    comment: |
+      This integration does not poll.
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: todo
```

Comment:
Oh but I am not mainly talking autodiscovery here, mostly about what is needed to create the port. For example for the tcp/ip one, I would maybe find the option to define `host` and `port` separately more logical. Since I can probably find the first one in my router and the last one in the documentation of my interface.

I am not that familiar with USB discovery, but I remember that there is another integration `energenie_power_sockets` and they list all the USB devices they find. Maybe that's also an option.

> at this moment the velbuslink (configuration) software will give you all the info to build the 'port' parameter

Keep in mind that not everyones network supports mDNS, so autodiscovery is fun, but not for everyone. In this case maybe a separate step would be worth it where you can take the time explaining the user how to generate that port using a description and a little walkthrough in the software.


And again, I am not a velbus expert, I am just trying to push back and challenge you a bit :). I think it would be nice if we could show a menu asking how the device is connected/what kind of device is it, where USB just lists all options, the tcp/ip one shows 2 input fields for host and port and signum shows a description on how to find the url in the software. (and then in the end I don't really care how we store it, we just have to keep in mind that you might want to implement a reconfigure flow, so in theory this can just be a config flow change and the rest of the intgration can stay roughly the same)

Reply:
yeah will do this in a follow-up PR, at this moment its still a TODO, so i know there is work to improve it